### ISSUE TITLE
passwordsettings check is not reliable

### ISSUE THREAD

**Issue by pdirksen** [kakwa/puppet-samba#4]
Since `samba-tool domain passwordsettings show -d 1` shows a different output depending on the samba version please add a 'grep filedname' (or something) to the checks.
This won't work with samba 4.2.

Default in Samba 4.2:

```
Password complexity: on
Store plaintext passwords: off
Password history length: 24
Minimum password length: 10
Minimum password age (days): 0
Maximum password age (days): 360
Account lockout duration (mins): 30
Account lockout threshold (attempts): 0
Reset account lockout after (mins): 30
```

**Comment by kakwa** [comment]
The module now provides a ppolicy_params resource which could be called to set individual ppolicy parameters.

The ppolicy class should now work with samba 4.2, but only handles parameters from samba 4.1. It could be completed with ppolicy_params resources.
Another possibility is to completely set the password policy with ppolicy_params resources.

It's not a very satisfying solution, but it's the best I could come up with.

**Comment by kakwa** [comment]
Fixed in release 0.3.0
